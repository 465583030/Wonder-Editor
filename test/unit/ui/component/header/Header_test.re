open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe(
    "Header ui component",
    () => {
      let sandbox = getSandboxDefaultVal();
      let _getFromArray = (array, index) => OperateArrayUtils.getNth(index, array);
      let _buildHeaderComponent = () =>
        ReactTestRenderer.create(
          <Header
            store=(SceneTreeToolUI.buildAppStateSceneGraphFromEngine())
            dispatch=(TestToolUI.getDispatch())
          />
        );
      let _buildEngineSceneTree = () =>
        ReactTestRenderer.create(
          <MainEditorSceneTree
            store=(SceneTreeToolUI.buildAppStateSceneGraphFromEngine())
            dispatch=(TestToolUI.getDispatch())
          />
        );
      let _buildInspectorComponent = () =>
        ReactTestRenderer.create(
          <MainEditorInspector
            store=(TestToolUI.buildEmptyAppState())
            dispatch=(TestToolUI.getDispatch())
            allShowComponentsConfig=(InspectorToolUI.buildFakeGameObjectComponentRecord())
          />
        );
      beforeEach(
        () => {
          sandbox := createSandbox();
          TestToolEngine.prepare(sandbox);
          TestToolUI.initMainEditor(sandbox);
          MainEditorSceneToolEditor.prepareDefaultScene(
            MainEditorSceneToolEditor.setFirstBoxTobeCurrentGameObject
          )
        }
      );
      afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
      describe(
        "test snapshot",
        () => {
          beforeEach(() => TestToolEditor.closeContractCheck());
          afterEach(() => TestToolEditor.openContractCheck());
          describe(
            "no operate gameObject",
            () => {
              test(
                "header ui component",
                () => {
                  let component = _buildHeaderComponent();
                  let json = ReactTestRenderer.toJSON(component);
                  toMatchSnapshot(expect(json))
                }
              );
              test(
                "sceneTree ui component",
                () => {
                  let component = _buildEngineSceneTree();
                  let json = ReactTestRenderer.toJSON(component);
                  toMatchSnapshot(expect(json))
                }
              );
              test(
                "inspector ui component",
                () => {
                  let component = _buildInspectorComponent();
                  let json = ReactTestRenderer.toJSON(component);
                  toMatchSnapshot(expect(json))
                }
              )
            }
          );
          describe(
            "operate gameObject",
            () => {
              beforeEach(
                () =>
                  MainEditorSceneToolEditor.unsafeGetCurrentGameObject()
                  |> MainEditorSceneToolEditor.addFakeVboBufferForGameObject
              );
              describe(
                "test add gameObject",
                () => {
                  let _triggerClickAddBox = (domChildren) => {
                    let addBoxDiv = _getFromArray(domChildren, 2);
                    let addBoxButton = _getFromArray(addBoxDiv##children, 0);
                    EventToolUI.triggerClickEvent(addBoxButton)
                  };
                  test(
                    "should only change sceneTree ui component",
                    () => {
                      let component = _buildHeaderComponent();
                      EventToolUI.triggerComponentEvent(component, _triggerClickAddBox);
                      let component2 = _buildEngineSceneTree();
                      let json = ReactTestRenderer.toJSON(component2);
                      toMatchSnapshot(expect(json))
                    }
                  )
                }
              );
              describe(
                "test dispose gameObject",
                () => {
                  let _triggerClickDispose = (domChildren) => {
                    let addBoxDiv = _getFromArray(domChildren, 3);
                    let addBoxButton = _getFromArray(addBoxDiv##children, 0);
                    EventToolUI.triggerClickEvent(addBoxButton)
                  };
                  test(
                    "change sceneTree ui component",
                    () => {
                      let component = _buildHeaderComponent();
                      EventToolUI.triggerComponentEvent(component, _triggerClickDispose);
                      let component2 = _buildEngineSceneTree();
                      let json = ReactTestRenderer.toJSON(component2);
                      toMatchSnapshot(expect(json))
                    }
                  );
                  test(
                    "change inspector ui component",
                    () => {
                      let component = _buildHeaderComponent();
                      EventToolUI.triggerComponentEvent(component, _triggerClickDispose);
                      let component2 = _buildInspectorComponent();
                      let json = ReactTestRenderer.toJSON(component2);
                      toMatchSnapshot(expect(json))
                    }
                  )
                }
              )
            }
          )
        }
      )
    }
  );