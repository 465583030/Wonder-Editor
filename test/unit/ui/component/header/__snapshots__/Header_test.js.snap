// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header ui component test snapshot no operate gameObject header ui component 1`] = `
<article
  className="header-component"
>
  <div
    className="component-item"
  >
    <button
      onClick={[Function]}
    >
      undo
    </button>
  </div>
  <div
    className="component-item"
  >
    <button
      onClick={[Function]}
    >
      redo
    </button>
  </div>
  <div
    className="component-item"
  >
    <button
      onClick={[Function]}
    >
      add box
    </button>
  </div>
  <div
    className="component-item"
  >
    <button
      onClick={[Function]}
    >
      dispose
    </button>
  </div>
  <div
    className="component-item"
  >
    <article
      className="wonder-file-input"
    >
      <button
        onClick={[Function]}
      >
        show Input
      </button>
    </article>
  </div>
</article>
`;

exports[`Header ui component test snapshot no operate gameObject inspector ui component 1`] = `
<article
  className="inspector-component"
>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        transform
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
    </article>
  </article>
  <article
    className="componentBox-component"
  >
    <div
      className="header"
    >
      <div
        className="header-triangle"
        onClick={[Function]}
      >
        <span
          className="triangle-bottom"
        />
      </div>
      <div
        className="header-title"
      >
        material
      </div>
    </div>
    <article
      className="transform-component"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          color : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="#ffffff"
        />
      </article>
    </article>
  </article>
  <article
    className="addable-component"
  >
    <button
      disabled={false}
      onClick={[Function]}
    >
      add component
    </button>
  </article>
</article>
`;

exports[`Header ui component test snapshot no operate gameObject sceneTree ui component 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "red",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject7
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`Header ui component test snapshot operate gameObject test add gameObject change sceneTree ui component 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "red",
          }
        }
      >
        gameObject5
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject7
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject8
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;

exports[`Header ui component test snapshot operate gameObject test dispose gameObject change inspector ui component 1`] = `
<article
  className="inspector-component"
/>
`;

exports[`Header ui component test snapshot operate gameObject test dispose gameObject change sceneTree ui component 1`] = `
<article
  className="sceneTree-component"
>
  <article
    className="wonder-drag-tree"
  >
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        camera
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject7
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
      draggable={true}
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <li
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        gameObject6
      </li>
    </ul>
    <div
      className="wonder-disable-drag"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "backgroundColor": "#c0c0c0",
        }
      }
    />
  </article>
</article>
`;
